tf-modules/cumulus_distribution/api.tf

```diff
@@ -158,3 +158,14 @@ resource "aws_api_gateway_deployment" "api" {
   stage_description = md5(file("${path.module}/api.tf"))
   stage_name        = var.api_gateway_stage
 }
+
+data "aws_lambda_invocation" "bucket_map_cache" {
+  function_name         = aws_lambda_function.api.function_name
+  input                 = jsonencode({
+    eventType           = "createBucketMapCache"
+    bucketList          = local.distribution_buckets,
+    s3Bucket            = var.system_bucket
+    s3Key               = "${var.prefix}/distribution_bucket_map.json"
```

Comment:
This seems like it should <do the same thing>, but... given right now it looks like cumulus-tf is deploying both this *and* TEA, is the intended behavior this will overwrite the tea generated file (which should be identical, in theory) or vice-versa?

Reply:
Perhaps more critically given the decoupled nature of the distribution module, perhaps the location should be a configurable option?       Meaning that right now we're using a hard-coded location in 

distribution-utils.ts :

```
export const getDistributionBucketMapKey = (stackName: string) =>
  `${stackName}/distribution_bucket_map.json`;
```

which should perhaps be set by a passed-in value to the Cumulus module such that external tf pacakge dependencies can provide it, so that *other* distribution options might be possible. 

(which.... also raises a question as to if we should have a good portion of these helper utilities for distribution in that package and/or if the distribution api really should be part of `api`, but exceeds scope here, admittedly).

Reply:
Since the bucket map file is created as part of the deployment of `distribution` or `cumulus_distribution` module, I I think we should keep the location internal and not configurable.

Reply:
>  yeah, cumulus_distribution and TEA will overwrite each other with the bucket map cache file

I'm not sure that's a releasable behavior, given TF likely doesn't know about the interdependence....  however I suppose if it's ordered and the distribution module is dependent on the Cumulus module via outputs, we could document that the new module overrides TEA.

Reply:
Added comment on bucket map cache [b6d7ae1](https://github.com/nasa/cumulus/pull/2322/commits/b6d7ae1ee722f5c2dfa9e8154e282dd3029ae2ae)
Ideally, the user should only deploy TEA or cumulus_distribution as their distribution endpoint.  Nate is working to change our example to use cumulus_distribution endpoint, in that case, `distribution` module is not deployed.
His PR:
https://github.com/nasa/cumulus/pull/2300
https://github.com/nasa/cumulus/pull/2300/files#diff-dece0af7ecf6c4c3f3856c5c14b567a5167db025c096bad9f90f4e186590d70aL1-R2

Reply:
Thanks! I think it's fine if @npauzenga updates example.    Is there any other documentation on the distribution API re: deployment yet?    I don't see an explicit ticket in https://bugs.earthdata.nasa.gov/browse/CUMULUS-2433, but I couldn't find deployment instructions. 

I'm less concerned about the dev team and more about users wanting *both* TEA and distribution (for reasons that users 'reason') in a scenario where they have a custom TEA bucketmap that doesn't map to what we're creating in the new `distribution` endpoint that could result in unexpected distribution updates. 

 Given we're releasing per sprint, merges to master need to be releasable as-is, and we should *explicitly* call that out in any deployment docs we might write.

Reply:
Approved, pending either a note in a ticket, update of docs or the concern being moot because TEA/Coffee can't be deployed together.

Reply:
Let me try to clarify the deployment plan. It's a little confusing.

>given right now it looks like cumulus-tf is deploying both this and TEA

That's actually not the case. The `distribution` module (which we use for TEA) will not be deployed if the user decides to use the new Cognito. See this PR:
https://github.com/nasa/cumulus/pull/2300/files#diff-dece0af7ecf6c4c3f3856c5c14b567a5167db025c096bad9f90f4e186590d70aR2

The situation where a user wants to have both TEA and Cumulus Distribution in a single deployment isn't really supported. You can switch between them (with downtime) by re-deploying with one or the other but you won't have 2 active distribution APIs available at once.

I have a documentation PR up that calls out the explicit switch between the two:
https://github.com/nasa/cumulus/pull/2353 

Also note that there's a difference between our `example` deployments and the `cumulus-template-deploy` deployments. `example` defaults to using the new Cumulus Distribution API via the assignments in `example/cumulus-tf/main.tf`:
https://github.com/nasa/cumulus/pull/2300/files#diff-192847ae3f8b97c1193324513f1a0e04d4f474bcf998b911a8a34370931c01c7R178

`cumulus_template_deploy` defaults to using TEA: https://github.com/nasa/cumulus-template-deploy/pull/63/files

Reply:
Thanks @npauzenga  - I'm fine approving this, but we can't release until the switching logic/docs are in then.     We can punt the discussion re: releasable PRs to other venues, as it's not directly specific to this PR.